### PR TITLE
Allow custom job name from passed job template

### PR DESCRIPTION
**Pull request by crflynn**
## Summary

This allows custom job name prefix using the job template passed to the KubernetesAgent.

## Changes

The current code in `KubernetesAgent.generate_job_spec` forces a job name prefix of `prefect-job`. This change  means that the job template's `metadata.name` (if it exists) will be used as the job name prefix. If not, it will fall back to `prefect-job`

## Importance

This PR allows users to name jobs so that they may be more easily identifiable.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Review by madkinsz — COMMENTED** [review]
@crflynn Thanks for contributing! Can you add a changes/ entry?

File: src/prefect/agent/kubernetes/agent.py [hunk 1]
```diff
@@ -479,7 +479,8 @@ def generate_job_spec(self, flow_run: GraphQLResult) -> dict:
 
         identifier = uuid.uuid4().hex[:8]
 
-        job_name = f"prefect-job-{identifier}"
+        job_name = _get_or_create(job, "metadata.name", "prefect-job")
```
Comment (jawnsy) [inline src/prefect/agent/kubernetes/agent.py:482]: If a user passes a `metadata.name` for a job, would they expect that the job name is used as-is? I suppose we have (at least) a few options here:

1. If a user passes a name, use it as-is (without any suffix) and if there is no name, then use the existing auto-generated `prefect-job-xxx` name.
2. We accept a "name_prefix" parameter when creating the job spec, which defaults to `prefect-job` and can be customized by the user (we'd append the identifier in this case, similar to how this PR behaves). Having this makes it clear that the name is a prefix for the job, but has the downside that we have to deal with the case where both a `metadata.name` and `name_prefix` are supplied. We can resolve that by using one preferentially to the other in that case.

Overall, I think I would find it surprising if I provided a job spec and the name was mutated in a way where I also could not disable or override that behavior.

Comment (madkinsz) [reply src/prefect/agent/kubernetes/agent.py:482]: We'll check with more of our team and see what makes the most sense.

Do you happen to have examples of the other schedulers that use this behavior? That'd be helpful.

Comment (crflynn) [reply src/prefect/agent/kubernetes/agent.py:482]: The other ones I've used are Airflow and Argo-workflows and IIRC they both add suffixes to jobs like prefect already does here.

Comment (crflynn) [reply src/prefect/agent/kubernetes/agent.py:482]: One other consideration here might be some defensive logic around the length of the identifier according to k8s resource name limits.

Comment (madkinsz) [reply src/prefect/agent/kubernetes/agent.py:482]: Do those pull the `metadata.name` from the job specification as the prefix though?

Comment (crflynn) [reply src/prefect/agent/kubernetes/agent.py:482]: * [yes](https://github.com/apache/airflow/blob/1bf6dded9a5dcc22238b8943028b08741e36dfe5/airflow/kubernetes/pod_generator.py#L126) for airflow
* [yes](https://github.com/argoproj/argo-workflows/blob/7e34ac5138551f0ebe0ca13ebfb4ad1fc8553ef1/workflow/common/convert.go#L37) for argo, although argo uses a CRD for its workflows but it pulls from the workflow name.

**Comment by crflynn** [comment]
Any movement on this?

**Review by madkinsz — APPROVED** [review]
Hey! I had some time to chat with more of the team about this and we're willing to accept it. Thank you for your patience! Can you add a brief blurb documenting how this works? Perhaps in https://docs.prefect.io/orchestration/agents/kubernetes.html#custom-job-template

**Comment by crflynn** [comment]
I will add something, sure. I also think we need to add some code which trims the name if it's too long. Not sure what the limits are but I'll add that as well.

**Pull request merged**